src/Agent.Listener/JobDispatcher.cs

```diff
@@ -464,7 +464,10 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, WorkerDisp
 
                     // we get first jobrequest renew succeed and start the worker process with the job message.
                     // send notification to machine provisioner.
-                    await notification.JobStarted(message.JobId);
+                    var systemConnection = message.Resources.Endpoints.SingleOrDefault(x => string.Equals(x.Name, WellKnownServiceEndpointNames.SystemVssConnection, StringComparison.OrdinalIgnoreCase));
+                    var accessToken = systemConnection?.Authorization?.Parameters["AccessToken"];
```

Comment:
What permission you need for publishing these telemetry?

Reply:
The access token is just used to initialize a VssClient and post to CI API. I don't think CI API needs any permission other than a valid token.